Model/Api/RouteInsuranceManagement.php

```diff
@@ -0,0 +1,148 @@
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api;
+
+use Bolt\Boltpay\Helper\Bugsnag;
+use Exception;
+use Magento\Framework\Webapi\Exception as WebapiException;
+use Magento\Framework\Webapi\Rest\Response;
+use Magento\Framework\Module\Manager;
+use Magento\Framework\Serialize\SerializerInterface as Serialize;
+use Bolt\Boltpay\Helper\Cart as CartHelper;
+use Magento\Quote\Model\QuoteRepository;
+
+class RouteInsuranceManagement implements \Bolt\Boltpay\Api\RouteInsuranceManagementInterface
+{
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var Manager
+     */
+    private $moduleManager;
+
+    /**
+     * @var Serialize
+     */
+    private $serializer;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var QuoteRepository
+     */
+    private $quoteRepository;
+
+    /**
+     * @param Response $response
+     * @param Bugsnag $bugsnag
+     * @param Manager $moduleManager
+     * @param Serialize $serializer
+     * @param CartHelper $cartHelper
+     * @param QuoteRepository $quoteRepository
+     */
+    public function __construct(
+        Response $response,
+        Bugsnag $bugsnag,
+        Manager $moduleManager,
+        Serialize $serializer,
+        CartHelper $cartHelper,
+        QuoteRepository $quoteRepository
+    ) {
+        $this->response = $response;
+        $this->bugsnag = $bugsnag;
+        $this->moduleManager = $moduleManager;
+        $this->serializer = $serializer;
+        $this->cartHelper = $cartHelper;
+        $this->quoteRepository = $quoteRepository;
+    }
+
+    /**
+     * @param $cartId
+     * @param $routeIsInsured
+     * @return void
+     * @throws WebapiException
+     */
+    public function execute($cartId, $routeIsInsured)
+    {
+        try {
+            if (!$this->isModuleEnabled()) {
+                $responseBody = $this->serializer->serialize(
+                    [
+                        'message' => sprintf("%s is not installed on merchant's site", self::ROUTE_MODULE_NAME)
+                    ]);
+                $httpResponseCode = self::RESPONSE_FAIL_STATUS;
+            } else {
+                $responseBody = $this->setRouteIsInsuredToQuote($routeIsInsured, $cartId);
+                $httpResponseCode = self::RESPONSE_SUCCESS_STATUS;
+            }
+            $this->responseBuilder($responseBody, $httpResponseCode);
+        } catch (WebapiException $e) {
+            $this->bugsnag->notifyException($e);
+            throw $e;
+        } catch (Exception $e) {
+            $this->bugsnag->notifyException($e);
+            throw new WebapiException(__($e->getMessage()), 0, WebapiException::HTTP_INTERNAL_ERROR);
+        }
+    }
+
+    private function setRouteIsInsuredToQuote($routeIsInsured, $cartId) {
```

Comment:
the function name is a little bit misleading because the function itself doesn't set something to quote.
Also, there are two purposes for the function: trigger quotes total collection and generate API answer.

I suggest splitting it into two parts and call the first one somethings like collectTotalsAndSave quote.
It makes sense to mention in comments why we want quotes to be recollected.